aws/resource_aws_launch_configuration.go

```diff
@@ -526,11 +527,26 @@ func resourceAwsLaunchConfigurationRead(d *schema.ResourceData, meta interface{}
 	d.Set("ebs_optimized", lc.EbsOptimized)
 	d.Set("spot_price", lc.SpotPrice)
 	d.Set("enable_monitoring", lc.InstanceMonitoring.Enabled)
-	d.Set("security_groups", lc.SecurityGroups)
 	d.Set("associate_public_ip_address", lc.AssociatePublicIpAddress)
+	if len(lc.SecurityGroups) > 0 {
```

Comment:
This can be simplified and error checked with:
```go
if err := d.Set("security_groups", flattenStringList(lc.SecurityGroups)); err != nil {
  return fmt.Errorf("error setting security_groups: %s", err)
}
```